addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -327,11 +327,11 @@ def _timesheet_preprocess(self, vals_list):
             if not vals.get('project_id'):
                 vals['project_id'] = data.project_id.id
             if not vals.get('account_id'):
-                account_id = data.analytic_account_id.id
-                account = account_per_id[account_id]
+                account = data._get_task_analytic_account_id() if vals.get('task_id') else data.analytic_account_id
                 if not account or not account.active:
                     raise ValidationError(_('Timesheets must be created on a project or a task with an active analytic account.'))
-                vals['account_id'] = account_id
+                account_per_id[account.id] = account
```

Comment:
Need to ensure the account `id` is known to avoid crash a few line below (when no `product_uom_id` in vals)

Reply:
Normally we should have the AA in that dict because we fill in that dict with the AA of the tasks (see line 308) and the projects (see line 313). 🤔 

cc @auon-odoo

Reply:
`account_ids.union(tasks.analytic_account_id.ids, tasks.project_id.analytic_account_id.ids)`
was good imo

Reply:
oups my bad, you are right 🙂

Reply:
update *again*, sorry for the back and forth :/

Reply:
No you're good, thanks for your time